libraries/AP_HAL_ChibiOS/EventSource.cpp

```diff
@@ -13,7 +14,7 @@ bool EventSource::wait(uint64_t duration, AP_HAL::EventHandle *evt_handle)
     if (duration == 0) {
```

Comment:
feels like you should do < CH_CFG_ST_TIMEDELTA here

Reply:
duration==0 is part of the API, so it needs to be a special case of no-wait